src/modules/file-upload-input.marko

```diff
@@ -0,0 +1,111 @@
+<div id="file-upload-input">
+    <section-header metadata=input.metadata name=input.name alpha="true"/>
+
+    <p>
+        The file-upload-input allows users to upload one or more files to a location. To make the entire component clickable and open the browse file dialog, a label is used to wrap the display elements and the input is hidden. The input element id should be${" "}
+        <span class="highlight">
+            file-upload-input__input
+        </span>
+         and the label for attribute should match the input id.
+    </p>
+    <p>
+        Apply the
+        <span class="highlight">
+            file-upload-input__upload-icon
+        </span>
+         class to set the upload icon.
+    </p>
+    <p>
+        Apply the
+        <span class="highlight">
+            file-upload-input__content
+        </span>
+        ${" "}class to set the content for the file-upload-input to provide upload instructions. This can include the header, subheader and a CTA button. If a button is defined, it should trigger the browse file dialog.
+    </p>
+    <p>
+        Apply the
+        <span class="highlight">
+            file-upload-input__content-header
+        </span>
+         class to set the content header.
+    </p>
+    <p>
+        Apply the
+        <span class="highlight">
+            file-upload-input__content-subheader
+        </span>
+         class to set the content subheader.
+    </p>
+    <p>
+        The component will change to a compact version for small screens at the medium breakpoint.
+    </p>
+
+    <h4>Drag and Drop</h4>
+    <p>
+        Since a label element is used to achieve the UI aesthetic and for a11y support, the drag and drop interaction will need to be added with JS on the label element.${" "}
+    </p>
+
+    <h3>Base</h3>
+
+    <div class="demo">
+        <div class="demo__inner">
+            <div class="file-upload-input" tabindex="0">
+                <label for="file-upload-input__input">
+                    <div class="file-upload-input__container">
+                        <div class="file-upload-input__upload-icon">
+                            <svg class="icon" aria-hidden="true">
+                                <use href="#icon-upload-24"/>
+                            </svg>
+                        </div>
+                        <div class="file-upload-input__content">
+                            <div class="file-upload-input__content-header">
+                                Add files
+                            </div>
+                            <div class="file-upload-input__content-subheader">
+                                Up to 10MB per file in JPG, JPEG, PNG, MP4.
+                            </div>
+                            <button class="fake-link" type="button">
```

Comment:
I don't think we can use a button here. The only nested interactive UI element should be the input type=file.